fairlearn/preprocessing/_correlation_remover.py

```diff
@@ -0,0 +1,85 @@
+# Copyright (c) Vincent D. Warmerdam, Microsoft Corporation and contributors.
+# Licensed under the MIT License.
+
+import numpy as np
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils import check_array
+from sklearn.utils.validation import check_is_fitted
+
+
+class CorrelationRemover(BaseEstimator, TransformerMixin):
```

Comment:
I personally like the name `CorrelationRemover`, but I wanted to bring back one alternative, which I think is a bit worse, but worth mentioning: `LinearDependenceRemover`.

Both names are accurate descriptions of the transformer. `CorrelationRemover` is more compact, but one possible disadvantage is that many people might think that it removes the **statistical dependence** (rather than just correlation). Calling it `LinearDependenceRemover` would deal with that... but the name is a bit awkward. Thoughts?

Reply:
I'm cool with either.

Reply:
I don't think `LinearDependenceRemover` is awkward, and it's a bit more explicit in saying what it does and that it's not a solution for all problems one could ever have. That said, this is not a hill I'm willing to die on :-)